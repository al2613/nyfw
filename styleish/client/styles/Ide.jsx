import styled from "styled-components";


//not reflecting the IDE. needs to be adjusted.
export const TranslateButton = styled.button`
  border: 1px solid black;
`;