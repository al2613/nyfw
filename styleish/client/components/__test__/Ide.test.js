import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import Ide from '../Ide.jsx';
import { TranslateButton } from '../../styles/Ide.jsx';

describe('Ide', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Ide />);
  });

  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('translate button get clicked', () => {
    const mockHandleSubmit = jest.fn();
    const button = shallow(<TranslateButton onClick={mockHandleSubmit} />);
    button.simulate('click');
    expect(mockHandleSubmit.mock.calls.length).toBe(1);
  });
});
