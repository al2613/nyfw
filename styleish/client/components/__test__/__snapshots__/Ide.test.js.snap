// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ide matches snapshot 1`] = `
<div>
  <SplitComponent
    cursorStart={1}
    editorProps={
      Object {
        "$blockScrolling": true,
      }
    }
    enableBasicAutocompletion={true}
    enableLiveAutocompletion={true}
    enableSnippets={true}
    focus={false}
    fontSize={12}
    height="250px"
    highlightActiveLine={true}
    maxLines={null}
    minLines={null}
    mode="css"
    name="app"
    onChange={[Function]}
    onLoad={null}
    onPaste={null}
    onScroll={null}
    orientation="beside"
    readOnly={false}
    scrollMargin={
      Array [
        0,
        0,
        0,
        0,
      ]
    }
    setOptions={Object {}}
    showGutter={true}
    showLineNumbers={false}
    showPrintMargin={true}
    splits={2}
    style={Object {}}
    tabSize={4}
    theme="xcode"
    value={
      Array [
        "/*Load your css*/ 
 
",
        "/*Styled component translation here*/ 
 
",
      ]
    }
    width="700px"
    wrapEnabled={true}
  />
  <styled.button
    onClick={[Function]}
  >
    TRANSLATE
  </styled.button>
</div>
`;
